### PR TITLE
fix function makeHunks

### PR DESCRIPTION
**Pull request by MHX123**
Resubmit a pr,Now only a line has been modified here.

### CODE REVIEW

File: hunk.js [hunk 1]
```diff
@@ -203,7 +203,7 @@ function makeHunks (changes, precontext, postcontext) {
         if (change.type === UNMODIFIED) {
             // add context
             let ctx_after  = ci > 0 ? postcontext : 0              // context lines following previous change
-            let ctx_before = ci < changes.length ? precontext : 0  // context lines preceding next change (iff there are more changes)
+            let ctx_before = ci < changes.length - 1? precontext : 0  // context lines preceding next change (iff there are more changes)
```
Comment (mvoss9000) [inline hunk.js]: Slightly strange to leave the 1 right next to the '?'

Comment (mvoss9000) [reply hunk.js]: Need to update the test.  The tests cover this case (and currently assert the wrong behavior.) test should be updated to this:

Screen Shot 2019-05-09 at 1 16 02 AM

Comment (mvoss9000) [reply hunk.js]: After making the update to the test, run the tests by going into the directory and doing:

npm install 
npm test

Make sense?

Comment (MHX123) [reply hunk.js]: Are you talking about the format?

Comment (MHX123) [reply hunk.js]: Hello,I have modified the format issue.

**Comment by mvoss9000** [comment]
I don't think it was clear.  I'm saying that the test needs to be updated with the change.  it's very simple.  Just run the test with 

    npm test

and you will see it.  can't fix the code and break the test.  Actually the test was asserting the WRONG behavior all along, so your fix shows how the test needs to be updated.

Just 3 lines in test/test_unidiff.js.   see it?  Update, run and pass tests and the pull request is good to go!

This update:

--- a/test/test_unidiff.js
+++ b/test/test_unidiff.js
@@ -227 +227 @@ test('unidiff: diffAsText - replace', function (t) {
-        '@@ -1,12 +1,12 @@',
+        '@@ -1,11 +1,11 @@',
@@ -242,2 +242 @@ test('unidiff: diffAsText - replace', function (t) {
-        ' line 11',
-        ' line 12'
+        ' line 11'

Make sense?

**Comment by mvoss9000** [comment]
LOL!

No Problem.

I was talking about the test!

Nobody is running the included tests!  Just try:

	npm test


I covers the case that you fixed.  So one of the assertions needed to be updated to match your fix (which is great by the way!).  I took care of it.  No problem.

Next time try the tests though - they should get updated together with the code to prove that it is working.

Thank you for your time though.  Appreciate your input!

 - Matthew

**Comment by VulumeCode** [comment]
Thanks a lot!

**Pull request merged**